src/aliceVision/image/cache.hpp

```diff
@@ -0,0 +1,333 @@
+  */
+  struct MRUItem
+  {
+    size_t objectId;
+    size_t objectSize;
+  };
+
+  /*
+  An item of the object to memory block associative array
+  */
+  struct MemoryItem
+  {
+    size_t startBlockId;
+    size_t countBlock;
+  };
+
+  using MemoryMap = std::map<size_t, MemoryItem>;
+
+  /**
+   Most recently used object container
+   Used to know which objects have not been used for some time
+   */
+  using MRUType = boost::multi_index::multi_index_container<
+                    MRUItem, 
+                    boost::multi_index::indexed_by<
+                      boost::multi_index::sequenced<>,
+                      boost::multi_index::hashed_unique<
+                        boost::multi_index::member<
+                          MRUItem, 
+                          size_t, 
+                          &MRUItem::objectId>
+                        >
+                      >
+                    >;
+
+public:
+  CacheManager() = delete;
+
+  /**
+   * Create a cache manager
+   * Each created object is associated to this manager.
+   * @param pathStorage the path to the directory where the file will be stored
+   * @param blockSize the base size of an object
+   * @param maxTilesPerIndex the maximal number of blocks for a given file (give a maximal size for a cache file)
+   */
+  CacheManager(const std::string & pathStorage, size_t blockSize, size_t maxBlocksPerIndex);
+  virtual ~CacheManager();
+
+  /**
+   * Set the maximal number number of items simultaneously in core
+   * @param max the maximal number of items
+   */
+  void setInCoreMaxObjectCount(size_t max);
+  
+  /**
+   * Create a new object of size block count
+   * @param objectId the created object index
+   * @param blockCount the required size for this object (In number of blocks)
+   * @return true if the object was created
+   */
+  bool createObject(size_t & objectId, size_t blockCount);
+
+  /**
+   * Acquire a given object
+   * @param data the result data acquired
+   * @param objectId the object index to acquire
+   * @return true if the object was acquired
+   */
+  bool acquireObject(std::unique_ptr<unsigned char> & data, size_t objectId);
+
+  /**
+   * Get the number of managed blocks
+   * @return a block count
+   */
+  size_t getActiveBlocks() const;
+
+protected:
+
+  std::string getPathForIndex(size_t indexId);
+  void deleteIndexFiles();
+  void wipe();
+
+  bool prepareBlockGroup(size_t startBlockId, size_t blocksCount);
+  std::unique_ptr<unsigned char> load(size_t startBlockId, size_t blocksCount);
+  bool save(std::unique_ptr<unsigned char> && data, size_t startBlockId, size_t blockCount);
+  bool saveObject(std::unique_ptr<unsigned char> && data, size_t objectId);
+
+  virtual void onRemovedFromMRU(size_t objectId) = 0;
+
+  void addFreeBlock(size_t blockId, size_t blockCount);
+  size_t getFreeBlockId(size_t blockCount);
+
+  
+
+protected:
+  size_t _blockSize;
+  size_t _incoreBlockUsageCount;
+  size_t _incoreBlockUsageMax;
```

Comment:
Add some doxygen doc to explain the CacheManager params.
For instance, I'm not sure what is _incoreBlockUsageMax.